pgx/src/lib.rs

```diff
@@ -319,4 +352,12 @@ macro_rules! pg_sql_graph_magic {
 #[allow(unused)]
 pub fn initialize() {
     pg_sys::panic::register_pg_guard_panic_hook();
+    match Lazy::force(&UTF8DATABASE) {
+        Utf8Compat::Maybe => notice!("database encoding is not validated as UTF-8"),
```

Comment:
There are encodings which can seem like UTF-8 for some number of bytes, and SQL_ASCII implies that we won't necessarily detect them before we complete some number of transactions. That's why the choice to emit a notice about this up-front.